internal/provider/resource_hvn_peering_connection.go

```diff
@@ -182,16 +180,15 @@ func resourceHvnPeeringConnectionCreate(ctx context.Context, d *schema.ResourceD
 
 func resourceHvnPeeringConnectionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*clients.Client)
-	orgID := client.Config.OrganizationID
 
-	link, err := buildLinkFromURL(d.Id(), PeeringResourceType, orgID)
+	link, err := buildLinkFromURL(d.Id(), PeeringResourceType, client.Config.OrganizationID)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
 	peeringID := link.ID
 	loc := link.Location
-	hvnLink1, err := buildLinkFromURL(d.Get("hvn_1").(string), HvnResourceType, orgID)
+	hvnLink1, err := buildLinkFromURL(d.Get("hvn_1").(string), HvnResourceType, loc.OrganizationID)
```

Comment:
The way I see it, it isn't usually important to know what the value of loc.OrganzationID is, as long as you know that it is the organization associated with the resource's unique link. If, for example, hvn1 must always use the same OrgID as the resource, referencing it with `loc.OrganizationID` (or `link.Location.OrganizationID`) provides an implicit guarantee that if the OrgID for the resource changes, that change will flow downstream to hvn1. 

From the other comment on this section of code, the relationship appears to be backward here, since the hvn1 `project_id` implies the resource's `project_id`.